README.md

```diff
@@ -128,87 +130,79 @@ Number matching is weak - the number has to appear
+the `Flattener` interface yourself.
 
 ## APIs
-
 **Note**: In all the APIs below, field names and values in both
 Patterns and Events must be valid UTF-8.  Unescaped characters
 smaller than 0x1F (illegal per JSON), and bytes with value
 greater than 0XF4 (can't occur in correctly composed UTF-8)
 are rejected by the APIs.
-
+### Control APIs
 ```go
-type Matcher interface {
-	AddPattern(x X, pat string) error
-	MatchesForJSONEvent(event []byte) ([]X, error)
-	MatchesForFields(fields []Field) []X
-	DeletePattern(x X) error
-}
-```
-
-Above are the operations provided by a Matcher. Quamina
-includes an implementation called `CoreMatcher` which
-implements `Matcher`.  In a forthcoming release it will
-provider alternate implementations that offer extra
-features.
+func New(...Option) (*Quamina, error)
 
-```go
-func NewCoreMatcher() *Matcher
+func WithMediaType(mediaType string) Option
+func WithFlattener(f Flattener) Option
+func WithPatternDeletion(b bool) Option
+func WithPatternStorage(ps LivePatternsState) Option 
 ```
+For example:
+
 ```go
-func pruner.NewMatcher() *Matcher
+q, err := quamina.New(quamina.WithMediaType("application/json"))
 ```
-
-Create new Matchers, take no arguments. The difference
-is that the `pruner.NewMatcher` version supports the
-`DeletePattern()` API. Be careful: It occasionally
-rebuilds the Matcher in stop-the-world fashion, so if you
-delete lots of Patterns in a large Matcher you may 
-encounter occasional elevated latencies.
+The meanings of the `Option` functions are:
+
+`WithMediaType`: In the futue, Quamina will support 
+Events not just in JSON but in other formats such as
+Avro, Protobufs, and so on. This option will make sure
+to invoke the correct Flattener. At the moment, the only
+supported value is `application/json`, the default.
+
+`WithFlattener`: Requests that Quamina flatten events with
+the provided (presumably user-written) Flattener.
+
+`WithPatternDeletion`: If true, arranges that Quamina
+allows Patterns to be deleted from an instance. This is 
+not free; it can incur extra costs in memory and 
+occasional stop-the-world Quamina rebuilds. (We plan
+to improve this.)
+
+`WithPatternStorage`: If you provide an argument that
+supports the `LivePatternStorage` API, Quamina will
+use it to 
+maintain a list of which patterns have currently been
+added but not deleted.  This could be useful if you
+wanted to rebuild Quamina instances for sharded 
+processing or after a system failure. ***Note: Not
+yet implemented.***
+
+### Data APIs
 
 ```go
-func (m *Matcher) AddPattern(x X, patternJSON string) error
+func (q *Quamina) AddPattern(x X, patternJSON string) error
 ```
-
 The first argument identifies the Pattern and will be
-returned by a Matcher when asked to match against Events.
-X is currently `interface{}`. Should it be a generic now
-that Go has them?
+returned by Quamina when asked to match against Events.
+X is defined as `any`.
 
 The Pattern must be provided as a string which is a 
```

Comment:
question: if I pick a different `Flattener`, why not call it `pattern`, i.e. make it not about JSON?

Reply:
Oops, need to be clearer. Right now we should be able to support Events in lots of different formats, once there are Flatteners.  But Patterns are just in JSON. I suppose in principle you could also support Patterns in protobufs etc but that wouldn't work well because Patterns are designed to be human-authored and humans can't read Avro. So for the moment JSON will do.